src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -0,0 +1,145 @@
+   * It then forwards to chat.jsp for rendering.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    // String requestUrl = request.getRequestURI();
+    // String conversationTitle = requestUrl.substring("/chat/".length());
+    //
+    // Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
+    // if (conversation == null) {
+    //   // couldn't find conversation, redirect to conversation list
+    //   System.out.println("Conversation was null: " + conversationTitle);
+    //   response.sendRedirect("/conversations");
+    //   return;
+    // }
+    //
+    // UUID conversationId = conversation.getId();
+    //
+    // List<Message> messages = messageStore.getMessagesInConversation(conversationId);
+    //
+    // request.setAttribute("conversation", conversation);
+    // request.setAttribute("messages", messages);
+    request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
+  }
+
+  /**
+   * This function fires when a user submits the form on the chat page. It gets the logged-in
+   * username from the session, the conversation title from the URL, and the chat message from the
+   * submitted form data. It creates a new Message from that data, adds it to the model, and then
+   * redirects back to the chat page.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String username = (String) request.getSession().getAttribute("user");
+    if (username == null) {
+      // user is not logged in, don't let them add a message
+      response.sendRedirect("/login");
+      return;
+    }
+
+    User user = userStore.getUser(username);
+    if (user == null) {
+      // user was not found, don't let them add a message
+      response.sendRedirect("/login");
+      return;
+    }
+
+    String requestUrl = request.getRequestURI();
+    String conversationTitle = requestUrl.substring("/chat/".length());
+
+    Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
+    if (conversation == null) {
+      // couldn't find conversation, redirect to conversation list
+      response.sendRedirect("/conversations");
+      return;
+    }
+
+    String messageContent = request.getParameter("message");
+
+    // this removes any HTML from the message content
+    String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
+
+    Message message =
+        new Message(
+            UUID.randomUUID(),
+            conversation.getId(),
+            user.getId(),
+            cleanedMessageContent,
+            Instant.now());
+
+    messageStore.addMessage(message);
+
+    // redirect to a GET request
+    response.sendRedirect("/chat/" + conversationTitle);
+  }
+}
```

Comment:
try to keep only the relevant code necessary to run your page. from what i can see, 90% of the code on this page have functionalities related to the chat page, not the activity feed.